services/identifier_service.py

```diff
@@ -85,6 +71,28 @@ def identify(self, repository: Repository, user_id: str) -> FirestoreRepo:
         self.data_service.batch_create_repo(repo)
         return repo
 
+    def _skip_node(self, node: ContentFile) -> bool:
+        if node.type == "file":
+            is_invalid_filename = (
+                    node.name.startswith(("_", "."))
+                    or node.name.endswith((".d.ts", ".d.js", ".min.js"))
+                    or re.search("test", node.name, re.IGNORECASE)
+                    or not re.match(self.include_pattern, node.name)
+            )
+
+            # 99000 bytes is ~20k tokens (depending on the model)
+            is_too_large = node.size > 99000
```

Comment:
If we use the mixtral tokenizer, we can have a more deterministic way of managing tokens.

Reply:
i would've liked to. But it would then load the entire file into memory. Using any tokenizer will need the actual file content. This acts as a sort of "sanity check". If the model really can't handle the tokens, the generation will fail.

The only drawback is we might miss out on files that just barely pass `99000` bytes and still are under the context limit. We can fine-tune this later and create separate values based on model capabilities.